packages/dynamite/dynamite_end_to_end_test/test/all_of_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:dynamite_end_to_end_test/all_of.openapi.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('ObjectAllOf', () {
+    final object = ObjectAllOf(
+      (final b) => b
+        ..attribute1AllOf = 'attribute1AllOfValue'
+        ..attribute2AllOf = 'attribute2AllOfValue',
+    );
+
+    final json = {
+      'attribute1-allOf': 'attribute1AllOfValue',
+      'attribute2-allOf': 'attribute2AllOfValue',
+    };
+
+    expect(object.toJson(), equals(json));
+    expect(ObjectAllOf.fromJson(json), equals(object));
+  });
+
+  test('OneObjectAllOf', () {
+    final object = OneObjectAllOf(
+      (final b) => b..attributeAllOf = 'attributeAllOfValue',
+    );
+
+    final json = {
+      'attribute-allOf': 'attributeAllOfValue',
+    };
+
+    expect(object.toJson(), equals(json));
+    expect(OneObjectAllOf.fromJson(json), equals(object));
+  });
+
+  test('PrimitiveAllOf', () {
+    final object = PrimitiveAllOf(
+      (final b) => b
+        ..string = 'stringValue'
+        ..$int = 0,
```

Comment:
0 might not be the best value because it is sometimes used as a default value. I'm not sure if that could ever happen here, but I'd rather avoid it from the beginning.